test/tests/functional/pbs_apbasil_inventory.py

```diff
@@ -0,0 +1,223 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from string import Template
+import os
+import tempfile
+import xml.etree.ElementTree as ET
+
+
+class TestBasilInventory(TestFunctional):
+    """
+    This test suite is for testing the support for BASIL 1.4 inventory
+    query.Test if query is made with correct BASIL version, and that
+    vnodes are getting created as per the query response.
+    """
+    basil_version = ['1.4', '1.3', '1.2', '1.1']
+
+    def restartPBS(self):
+        try:
+            svcs = PBSInitServices()
+            svcs.restart()
+        except PbsInitServicesError, e:
+            self.logger.error("PBS restart failed: \n" + e.msg)
+            self.assertTrue(e.rv)
+
+    def reset_nodes(self, hostA):
+        # Remove all nodes
+        rv = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rv, 0)
+
+        # Restart PBS
+        self.restartPBS()
+
+        # Create node
+        rv = self.server.manager(MGR_CMD_CREATE, NODE, None, hostA)
+        self.assertEqual(rv, 0)
+
+        # Wait for 3 seconds for changes to take effect
+        time.sleep(3)
+
+    def setup(self):
+        momA = self.moms.values()[0]
```

Comment:
I think you don't need self.moms.values()[0] here as the init_moms function of PBSTestSuite is doing this
[here is the link](https://github.com/PBSPro/pbspro/blob/master/test/fw/ptl/utils/pbs_testsuite.py#L690)

Why the function name is 'setup' (not 'setUp')?

Other than above comments, the code looks good to me.